opentelemetry/src/main/scala/zio/telemetry/opentelemetry/tracing/Tracing.scala

```diff
@@ -685,17 +691,22 @@ object Tracing {
             getCurrentSpan.map(_.setAttribute(AttributeKey.doubleArrayKey(name), v))
           }
 
-          private def setErrorStatus[E](
+          override def setErrorStatus[E](
             span: Span,
             cause: Cause[E],
-            toErrorStatus: ErrorMapper[E]
+            errorMapper: ErrorMapper[E]
           )(implicit trace: Trace): UIO[Span] = {
             val errorStatus =
               cause.failureOption
-                .flatMap(toErrorStatus.body.lift)
+                .flatMap(errorMapper.body.lift)
                 .getOrElse(StatusCode.UNSET)
 
-            ZIO.succeed(span.setStatus(errorStatus, cause.prettyPrint))
+            for {
+              spanWithStatus <- ZIO.succeed(span.setStatus(errorStatus, cause.prettyPrint))
+              result         <- errorMapper.toThrowable.fold(ZIO.succeed(spanWithStatus))(toThrowable =>
+                                  ZIO.succeed(spanWithStatus.recordException(cause.squashWith(toThrowable)))
+                                )
+            } yield result
```

Comment:
Since `span.setStatus` is mutable, we might reuse `span` in the code below instead of introducing an additional `spanWithStatus` variable. Example:
```
            for {
              _      <- ZIO.succeed(span.setStatus(errorStatus, cause.prettyPrint))
              result <- errorMapper.toThrowable.fold(ZIO.succeed(span))(toThrowable =>
                          ZIO.succeed(span.recordException(cause.squashWith(toThrowable)))
                        )
            } yield result
```